packages/console/src/pages/OneTimeTokenLanding/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { useLogto } from '@logto/react';
+import { ExtraParamsKey } from '@logto/schemas';
+import { useContext, useEffect } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import AppLoading from '@/components/AppLoading';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import useRedirectUri from '@/hooks/use-redirect-uri';
+
+enum OneTimeTokenLandingSearchParams {
+  OneTimeToken = 'one_time_token',
+  Email = 'email',
+}
+
+/** The one-time token landing page for sign-in with one-time tokens. */
+function OneTimeTokenLanding() {
+  const navigate = useNavigate();
+  const { isAuthenticated, signIn, isLoading } = useLogto();
+  const [searchParams] = useSearchParams();
+  const { navigateTenant, currentTenantId } = useContext(TenantsContext);
+  const redirectUri = useRedirectUri();
+
+  useEffect(() => {
+    if (isAuthenticated) {
+      // If we have a current tenant, navigate to it
+      if (currentTenantId) {
+        navigateTenant(currentTenantId);
+      } else {
+        // Otherwise navigate to root, which will handle tenant selection
+        navigate('/', { replace: true });
+      }
+    }
+  }, [isAuthenticated, navigate, currentTenantId, navigateTenant]);
```

Comment:
This `useEffect` can be saved if you merge the condition check in the second useEffect hook. And you don't need to check the currentTenantId, as the root element should be able to figure it out itself.